collections/indexes/multi.go

```diff
@@ -121,6 +123,41 @@ func (m *Multi[ReferenceKey, PrimaryKey, Value]) MatchExact(ctx context.Context,
 	return m.Iterate(ctx, collections.NewPrefixedPairRange[ReferenceKey, PrimaryKey](refKey))
 }
 
+// RefKeys returns a list of all the MultiIterator's reference keys (may contain duplicates).
+// Enable the "unique" argument to get a unique list of reference keys (the reference key must be comparable)
+// WARNING: The use of RefKeys() can be very expensive in terms of Gas. Please make sure you iterate over a relatively
+// small set of reference keys.
+func (m *Multi[ReferenceKey, PrimaryKey, Value]) RefKeys(ctx context.Context, unique bool) ([]ReferenceKey, error) {
+	iter, err := m.refKeys.IterateRaw(ctx, nil, nil, collections.OrderAscending)
+	if err != nil {
+		return nil, err
+	}
+
+	keys := []ReferenceKey{}
+	visited := map[[32]byte]struct{}{}
+	for ; iter.Valid(); iter.Next() {
+		key, err := iter.Key()
+		if err != nil {
+			return nil, err
+		}
+
+		if unique {
+			// compare the byte representation of ref keys
+			refKey := key.K1()
+			unsafeRefKey := *(*[]byte)(unsafe.Pointer(&refKey))
+
+			// use SHA256 hash as map keys
+			if _, ok := visited[sha256.Sum256(unsafeRefKey)]; ok {
```

Comment:
we should not hash, at best (or worst) we should use just string(x)

Reply:
I am now using `string()` instead of hash. see https://github.com/oren-lava/cosmos-sdk/commit/551ed3475e3b532544b0a2b054712882cedaf824